src/main/java/htsjdk/samtools/util/IOUtil.java

```diff
@@ -238,6 +237,21 @@ public static void deleteFiles(final Iterable<File> files) {
         }
     }
 
+    public static void deletePaths(final Path... paths) {
+        deletePaths(Arrays.asList(paths));
+    }
+
+    public static void deletePaths(final Iterable<Path> paths) {
+        for (final Path p : paths) {
+            boolean deleted;
+            try {
+                deleted = Files.deleteIfExists(p);
+            } catch (IOException e) {
+                deleted = false;
+            }
+            if (!deleted) System.err.println("Could not delete file " + p);
```

Comment:
Why not use `Files.delete()` if you want to log to stderr in the case where the file doesn't exist?

Reply:
I tried my best to make parallel the methods of `File` and `Path`, and I forgot things like this. Changed to `Files.delete()` to catch also non-existence.